### DISCUSSION TITLE
How to add AES GCM with bcrypt for .NET framework?

### DISCUSSION THREAD

**Discussion by pavinan** [microsoft/CsWin32#1144] [Q&A] [answered]
I installed this CsWin32 package in netstandard2.0 class library. But it is giving error to use c#9.0 to support `nint` keyword. Is it it not possible to use aesgcm with this in .net framework?

I have used BCRYPT* in that NativeMethods.txt

**Comment by AArnott** [comment, marked as answer]
Down-level C# language version support is spotty. You should be able to fix this by adding this to your project file:

```xml
<LangVersion>9</LangVersion>
```

Of course, I like to go all the way and use 12. 

C# 12 has the appropriate guards to keep you on the right path when targeting older runtimes.

**Reply by pavinan** [reply]
But .netstandard supports only c# 7.3 right? But I will check if it works with 9 or 12.

**Reply by AArnott** [reply]
.NET Standard doesn't support any _language_ at all. It's a set of ref assemblies. Runtimes also don't support any particular language. The language is only a means to generate ecma-335 assemblies, which C# 7.3 and C# 12 both do. 
So yes, you absolutely can use C# 12 when targeting .NET Standard 2.0.
